tools/use.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+: <<EOF
+
+Source this script to import components into other scripts. Usage:
```

Comment:
I had typed up a comment here and GitHub dropped it arggghh

---

I was tempted to go the "import each file as you need it" route, but settled on this additional complexity for a few reasons, in order of least-to-most helpful:

- Catch cyclic imports immediately
- Reduce the need for path-juggling to import each component separately
- Have a one-liner at the top that specifies exactly which components are being used
- Have an `is-main` utility: This is especially useful to use the scripts as command-line tools on their own. I use our bash scripts regularly as part of development workflow, and making it as convenient as possible to run things identically to the process in CI is very helpful. (This is also why I put them in `tools/` instead of `.evergreen/`.)

As for explicit-defining/sourcing, I think that's mostly a matter of personal preference, where I prefer a single-source-of-truth for variables and utilities, even if it means I pull additional components in when I only need a subset of them. (Using single-truths was extremely helpful to remove redundant and subtly-broken/different path juggling and util functions when refactoring the libmongocrypt tree to be CWD-agnostic.)

Reply:
The motivation and rationale is greatly appreciated. Agreed regarding personal preference. Thank you for bearing with my conflicting perspectives in this PR.